platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.error.injector;
+
+import com.google.common.collect.ImmutableMap;
+import java.lang.reflect.Field;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.ServletContext;
+import org.codice.ddf.platform.error.servlet.ErrorServlet;
+import org.eclipse.jetty.servlet.ErrorPageErrorHandler;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ErrorPageInjector implements EventListenerHook {
+
+  private final String errorPagePath = "/ErrorServlet";
+  private ImmutableMap<String, String> errorCodesMap =
+      new ImmutableMap.Builder<String, String>()
+          .put("400", errorPagePath)
+          .put("401", errorPagePath)
+          .put("403", errorPagePath)
+          .put("404", errorPagePath)
+          .put("405", errorPagePath)
+          .put("406", errorPagePath)
+          .put("500", errorPagePath)
+          .put("501", errorPagePath)
+          .put("502", errorPagePath)
+          .put("503", errorPagePath)
+          .put("504", errorPagePath)
+          .build();
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ErrorPageInjector.class);
+
+  Optional<BundleContext> getContext() {
```

Comment:
✏️ Could rename this method to clarify what bundle context you are grabbing, maybe something like `getErrorPageInjectorContext`.

Reply:
❓ ✏️ Also, could blueprint make getting the `ErrorPageInjector` class easier?

Reply:
talked to jake 👍